tests/cli/test_cli.py

```diff
@@ -164,7 +164,7 @@ def get_info(uuid, key):
     return _run_command([cl, 'info', '-f', key, uuid])
 
 
-def wait_until_state(uuid, expected_state, timeout_seconds=1000):
+def wait_until_state(uuid, expected_state, timeout_seconds=1000, exclude_final_states=set()):
```

Comment:
What exactly does "exclude_final_states" mean? Can you document in the docstring?

Reply:
Yes. Essentially, I use this because some states aren't actually 'final'; in this case, the bundle enters State.FAILED before entering State.KILLED. The reason is that, after the bundle finishes, the worker attempts to upload the bundle to the bundle store, but that upload fails since the user doesn't have enough quota (so the bundle state is updated to State.FAILED). However, the bundle state is updated to State.KILLED soon after that.

Essentially, then, in this case, the State.FAILED is not the final state. There are a few ways to go about testing this:
- make it so we can test that the bundle state _eventually_ reaches State.KILLED (what I do here)
- change how the bundle state is actually updated (i.e. modify it so that it doesn't update to State.FAILED at all)
- just call wait_until_state and set expected_state to State.FAILED

I chose the first option because, by waiting for it to be killed, I could also check that the kill message is what I expected it to be.